PL2/PL2-shared/src/main/java/nl/tudelft/pl2016gr2/model/OriginalGraph.java

```diff
@@ -80,6 +124,10 @@ public Node getRoot() {
     return nodes.get(lowestId);
   }
 
+  /**
+   * @deprecated Use {@link #iterator()} instead. This method exposes the hashmap.
+   * {@inheritDoc}
+   */
```

Comment:
Iterators don't give any way to perform the action: nodes.contains(key) efficiently, so this would have to be implemented by this class if you want to remove this method.